shared-libs/user-management/src/users.js

```diff
@@ -770,8 +786,18 @@ const getUserSettings = async({ name }) => {
  */
 module.exports = {
   deleteUser: username => deleteUser(createID(username)),
-  getList: async () => {
-    const [ users, settings ] = await Promise.all([ getAllUsers(), getAllUserSettings() ]);
+  getList: async (filters) => {
+    let users;
+    let settings;
+
+    if (_.isEmpty(filters)) {
+      [users, settings] = await Promise.all([getAllUsers(), getAllUserSettings()]);
+    } else {
+      settings = await getAllUserSettings(filters);
+      const ids = settings.map(({ _id }) => _id);
+      users = await getUsersByIds(ids);
```

Comment:
User docs have a `facility_id` property but no `contact_id`, that's why we have to query the `user-setting` docs first and then get the user docs from their ids

Reply:
Nice and efficient to just get the users by id!  

One thing I do want to call out here is that we are filtering by the `facility_id` in the `user-settings` doc (and not the one in the `_users` doc).  This has the [acknowledged drawback](https://github.com/medic/cht-core/issues/8877#issuecomment-1952052847) that currently users can edit their own `user-settings` doc.  Sounds like this will be locked down in the future.  (Also, this may be the reason this code is not running into the issues with non-admin users calling the ` /_users/_find` endpoint....)

I don't think we need to change anything here (any change I can think of will just be worse performance), unless we think it is important that the API code uses the `facility_id` data from the `_users` doc...

Reply:
@m5r did you happen to get @garethbowen's input on using the `user-settings.facility_id` to query on?  The more I think about this, the more I would like to know his thoughts before shipping this.  (But he is OOO all of next week....)